plugins/catalog/src/components/CatalogPage/CatalogPage.tsx

```diff
@@ -48,12 +48,18 @@ export type CatalogPageProps = {
   initiallySelectedFilter?: UserListFilterKind;
   columns?: TableColumn<EntityRow>[];
   actions?: TableProps<EntityRow>['actions'];
+  createButton?: boolean;
+  supportButton?: boolean;
+  createButtonTitle?: string;
```

Comment:
This is an unfortunate slippery slope of page customization. Introducing props for various components on the page will eventually create a bit of a mess. The [catalog customization](https://backstage.io/docs/features/software-catalog/catalog-customization) page shows how you customize this page without too much effort. That's not 100% ideal either, since although the JSX is designed to be minimal - you still end up making a copy of it.

Composability of Backstage pages is a broad topic that's on the horizon, since swapping out components and customizing UI is a topic that gets hit upon often. There should ideally be a generic mechanism to do things like this, which applies broadly across the codebase instead of one-off prop changes for particular needs.

In the meantime, for this particular need, I think perhaps the [catalog customization](https://backstage.io/docs/features/software-catalog/catalog-customization) guidance is the way to go. I'm curious to hear thoughts from @backstage/maintainers though.

Reply:
Yep - I'm not sure that adding more and more props here is the best way to do this, as we just keep adding things to an every increasing and ever more brittle public API. Much rather do this with the existing work if we can.

Reply:
Makes sense, guys. I was just trying something, but I guess a customizable CatalogPage will have to do

Reply:
I will close this PR @Aj-vrod, but if you need to extend the functionality in the [catalog customization](https://backstage.io/docs/features/software-catalog/catalog-customization) work, then please open another PR with some work over there. Happy help too if you want to just open an issue for some guidance :)